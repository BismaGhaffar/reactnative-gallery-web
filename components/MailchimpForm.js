import styled from 'styled-components'
import Hideable from '../components/Hideable'
import { Share, Check } from 'react-feather'

const MailchimpForm = styled.form`
  margin: 0 auto;
  width: 100%;
  max-width: 600px;
  display: flex;
  flex-direction: row;
  padding-top: 30px;
  padding-bottom: 30px;
`

export const MailchimpInput = styled.input`
  border-width: 0;
  border-radius: 0;
  font-size: 14px;
  height: 50px;
  display: block;
  width: 100%;
  height: 36px;
  padding: 6px 12px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
`

export const MailchimpButton = styled.button`
  display: block;
  margin: 0 auto;
  border-width: 0;
  border-radius: 0;
  font-size: 14px;
  padding-right: 20px;
  padding-left: 20px;
  height: 50px;
  background-color: #00a651;
  color: #fff;
  line-height: 0.5em;
`

export default ({ action, type }) => (
  <MailchimpForm name="form" noValidate="" action={action} method="POST" target="_blank">
    <MailchimpInput
      type="email"
      name="EMAIL"
      placeholder="Email"
      ariaRequired="true"
      required="required"
    />

    <MailchimpButton type="submit">
      <Hideable xs>SUBMIT</Hideable>
      <Hideable md>
        <Check />
      </Hideable>
    </MailchimpButton>
    <input type="hidden" name="TYPE" value={type} />
  </MailchimpForm>
)
