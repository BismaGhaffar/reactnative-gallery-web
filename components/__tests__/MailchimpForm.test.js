import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MailchimpForm from '../MailchimpForm'

Enzyme.configure({ adapter: new Adapter() })

it('MailchimpForm can be created', () => {
  const comp = renderer.create(<MailchimpForm
    action="test"
    type="test"
    email="test@email.com"
    onChange={() => {}}
  />)
  expect(comp).toBeDefined()
})

it('<MailchimpForm /> toMatchSnapshot', () => {
  const tree = renderer
    .create(<MailchimpForm
      action="test"
      type="test"
      email="test@email.com"
      onChange={() => {}}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('<MailchimpForm /> simulate input change ', () => {
  // TODO: find a way to pass intno onChange
  const email = 'cool@gmel.fr'
  const form = shallow(<MailchimpForm
    action="test"
    type="test"
    email={email}
    onChange={(e) => {
        console.log(e)
      }}
  />)

  form.simulate('change', { target: { value: 'cool@gmel.fr' } })
  expect(form.childAt(1).props().children[0].props.value).toBe('cool@gmel.fr')
})
