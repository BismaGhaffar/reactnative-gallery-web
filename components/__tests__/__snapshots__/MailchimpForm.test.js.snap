// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MailchimpForm /> toMatchSnapshot 1`] = `
<div>
  <div
    className="CleanHr-u75eti-0 fnhkPK"
  />
  <p
    className="Paragraph__Subtitle-s1secd56-0 iVQIYc"
    style={
      Object {
        "margin": "auto",
        "marginTop": 60,
        "maxWidth": 550,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Enter your email address if you want to be informed when it's ready
  </p>
  <form
    action="test"
    className="MailchimpForm-s1jd0qpd-0 gwLiEU"
    method="POST"
    name="form"
    noValidate=""
    target="_blank"
  >
    <input
      className="MailchimpForm__MailchimpInput-s1jd0qpd-1 hkXAOt"
      name="EMAIL"
      placeholder="Email"
      required="required"
      type="email"
    />
    <button
      className="MailchimpForm__MailchimpButton-s1jd0qpd-2 bHDbCA"
      type="submit"
    >
      <span
        className="Hideable-lb4mgy-0 bDYHzV"
      >
        SUBMIT
      </span>
      <span
        className="Hideable-lb4mgy-0 UfVDy"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </span>
    </button>
    <input
      name="TYPE"
      type="hidden"
      value="test"
    />
  </form>
</div>
`;
